tools/reckless

```diff
@@ -0,0 +1,684 @@
+        if not confirm:
+            sys.exit(1)
+        parent_path = os.path.split(config_path)[0]
+        # Create up to one parent in the directory tree.
+        if create_dir(1, parent_path):
+            with open(self.conf_fp, 'w') as f:
+                f.write(default_text)
+                # FIXME: Handle write failure
+                return default_text
+        else:
+            print('could not create the parent directory')
+            return None
+
+    def editConfigFile(self, addline, removeline):
+        remove_these_lines = []
+        with open(self.conf_fp, 'r') as reckless_conf:
+            original = reckless_conf.readlines()
+            empty_lines = []
+            for n, l in enumerate(original):
+                if l.strip() == removeline:
+                    remove_these_lines.append(n)
+                    continue
+                if l.strip() == '':
+                    empty_lines.append(n)
+                    if n-1 in empty_lines:
+                        # The white space is getting excessive.
+                        remove_these_lines.append(n)
+                        continue
+            with open(self.conf_fp, 'w') as conf_write:
+                # no need to write if passed 'None'
+                line_exists = not bool(addline)
+                for n, l in enumerate(original):
+                    if n not in remove_these_lines:
+                        if n > 0:
+                            conf_write.write(f'\n{l.strip()}')
+                        else:
+                            conf_write.write(l.strip())
+                        if addline == l:
+                            # addline is idempotent
+                            line_exists = True
+                if not line_exists:
+                    conf_write.write(f'\n{addline}')
+
+    def __init__(self, path=None, default_text=None, warn=False):
+        assert path is not None
+        assert default_text is not None
+        self.conf_fp = path
+        self.content = self.obtain_config(self.conf_fp, default_text,
+                                          warn=warn)
+
+
+class RecklessConfig(Config):
+    """Reckless config (by default, specific to the bitcoin network only.)
+    This is inherited by the lightningd config and contains all reckless
+    maintained plugins."""
+
+    def enable_plugin(self, plugin_path):
+        """Handle persistent plugin loading via config"""
+        self.editConfigFile(f'plugin={plugin_path}',
+                            f'disable-plugin={plugin_path}')
+
+    def disable_plugin(self, plugin_path):
+        """Handle persistent plugin disabling via config"""
+        self.editConfigFile(f'disable-plugin={plugin_path}',
+                            f'plugin={plugin_path}')
+
+    def __init__(self, path=None, default_text=None):
+        if path is None:
+            path = os.path.join(LIGHTNING_DIR, 'reckless',
+                                'bitcoin-reckless.conf')
```

Comment:
You didn't fall into the mutable-default-argument trap that makes life of a pythonista interesting, well done :+1:

https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments